domehunter/dome_control.py

```diff
@@ -0,0 +1,674 @@
+        includes,
+        - A digital input device for the encoder (`ENCODER_PIN_NUMBER`)
+        - A digital input device for the home sensor (`HOME_SENSOR_PIN_NUMBER`)
+        - A digital output device for the motor on/off relay switch
+          (`ROTATION_RELAY_PIN_NUMBER`)
+        - A digital output device for the motor direction (CW/CCW) relay switch
+          (`DIRECTION_RELAY_PIN_NUMBER`)
+
+        The Digital Input Devices (DIDs) have callback functions that can be
+        used to call a function upon activation or deactivation  of a sensor
+        via a GPIO. For example, activating the encoder DID can be set to
+        increment a Dome encoder count instance variable.
+
+        As part of initialisation several instance variables will be set to
+        designate status infomation about the dome such as dome azimuth
+        (unknown at initialisation) and position of the direction relay switch
+        (initialised in the CCW position).
+
+        Parameters
+        ----------
+        testing : boolean
+            Toggle to enable a simulated hardware testing mode.
+        debug_lights : boolean
+            Toggle to enable the status LEDs on the automationHAT.
+        ENCODER_PIN_NUMBER : int
+            The GPIO pin that corresponds to the encoder input on the
+            automationHAT (input 1).
+        HOME_SENSOR_PIN_NUMBER : int
+            The GPIO pin that corresponds to the home input on the
+            automationHAT (input 2).
+        ROTATION_RELAY_PIN_NUMBER : int
+            The GPIO pin that corresponds to the rotation relay on the
+            automationHAT (relay 1). The Normally Open (NO) relay position
+            corresponds to clockwise (CW) and the Normally Closed (NC)
+            position corresponds to counterclockwise (CCW).
+        DIRECTION_RELAY_PIN_NUMBER : int
+            The GPIO pin that corresponds to the direction relay on the
+            automationHAT (relay 2).
+        BOUNCE_TIME : float
+            A buffer period (in seconds) where home/encoder input will ignore
+            additional (de)activation.
+
+        """
+
+        if testing:
+            # Set the default pin factory to a mock factory
+            Device.pin_factory = MockFactory()
+            # in case a previous instance has been initialised, tell the
+            # pin factory to release all the pins
+            Device.pin_factory.reset()
+
+            # set a timeout length in seconds for wait_for_active() calls
+            WAIT_TIMEOUT = 1
+
+            # in testing mode we need to create a seperate pin object so we can
+            # simulate the activation of our fake DIDs and DODs
+            self.encoder_pin = Device.pin_factory.pin(ENCODER_PIN_NUMBER)
+            self.home_sensor_pin = Device.pin_factory.pin(
+                HOME_SENSOR_PIN_NUMBER)
+        else:
+            # set the timeout length variable to None for non testing mode
+            WAIT_TIMEOUT = None
```

Comment:
Seems like we should have some high value for timeouts so we can handle potentially moving forever. Some similar comments below. Can be left for future.